### PR TITLE
Added sortition db schema transition logic

### PR DESCRIPTION
**Pull request by pavitthrap**
Addresses #2892 - adds a transition function for the sortition DB schema. 

Ran existing tests, and they passed.

### CODE REVIEW

File: src/chainstate/burn/db/sortdb.rs [hunk 1]
```diff
@@ -2267,6 +2377,37 @@ impl SortitionDB {
             .optional()?;
         Ok(version)
     }
+
+    fn apply_schema_2(tx: &SortitionDBTx) -> Result<(), db_error> {
+        for sql_exec in SORTITION_DB_SCHEMA_2 {
+            tx.execute_batch(sql_exec)?;
+        }
+
+        tx.execute("INSERT INTO db_config (version) VALUES (?1)", &["2"])?;
```
Comment (jcnelson) [inline src/chainstate/burn/db/sortdb.rs:2386]: Should be `INSERT OR REPLACE INTO`

File: src/chainstate/burn/db/sortdb.rs [hunk 2]
```diff
@@ -597,14 +597,15 @@ const SORTITION_DB_INITIAL_SCHEMA: &'static [&'static str] = &[
     );"#,
     "CREATE INDEX canonical_stacks_blocks ON canonical_accepted_stacks_blocks(tip_consensus_hash,stacks_block_hash);",
     "CREATE TABLE db_config(version TEXT NOT NULL);",
-    r#"
+];
```
Comment (jcnelson) [inline src/chainstate/burn/db/sortdb.rs:600]: There should only ever be one row in `db_config`, and it should be the current version.  Can you make the schema for `db_config` to be `db_config(version TEXT PRIMARY KEY)` instead, and use `INSERT OR REPLACE` when updating the version?

**Review by jcnelson — APPROVED** [review]
LGTM; just had two easily-addressable comments about making sure there's only ever one version row in `db_config`.

**Review by kantai — APPROVED** [review]
LGTM -- just address the `insert or replace into` feedback before merging.

**Pull request merged**